lib/completion.dart

```diff
@@ -146,6 +145,10 @@ class AnalysisCompletion implements Comparable {
 
   int get selectionOffset => _int(_map['selectionOffset']);
 
+  // FUNCTION, GETTER, CLASS, ...
+  String get type => _map['element']['kind'];
```

Comment:
Not all of these completions will have `element` submaps; you'll want to guard against `null`. Something like `_map['element'] == null ? null : _map['element']['kind']`.